### PR TITLE
.draftignore file support

### LINKED ISSUE #9: implement .draftignore
On `draft up` with watch enabled, there are files potentially being autogenerated in the background by dev tooling that should not trigger a `draft up` when the file changes. For now there are a few hardcoded file extensions that we ignore, but it'd be better to give the power back to the user by writing out a .draftignore to the app directory on `draft create`, which can be further modified.

### PR DESCRIPTION
**Pull request by didrocks**
Add `.draftignore` support using the helm ignore syntax (and package).
We scaffold a default ignore file on `draft create`.
Add some tests to cover those new use cases as well.

This PR builds upon https://github.com/Azure/draft/pull/146

Fixes #9

### CODE REVIEW

File: cmd/draft/up.go [hunk 1]
```diff
@@ -194,3 +212,29 @@ func defaultDraftEnvironment() string {
 	}
 	return env
 }
+
+func parseIgnores() (*ignore.Rules, error) {
+	// default ignored extensions
+	defaultsIR := strings.NewReader(strings.Join(defaultIgnores, "\n"))
```
Comment (bacongobbler) [inline cmd/draft/up.go]: I would prefer if we wrote out a `.draftignore` with these default ignores in the file. That way it's less "black magic" and users can actually check for these files if they wanted to.

**Comment by bacongobbler** [comment]
this is a fantastic PR to wake up to! :)

File: cmd/draft/up.go (hunk 1 above)
Comment (didrocks) [inline cmd/draft/up.go]: Yes! I was wondering which approach you would prefer, so I first did the more complex one. No worry, I just changed it to do this.

**Comment by didrocks** [comment]
Thanks for your quick feedback! I rebased the branch on latest master and did implement your suggestions, changing tests as needed.
`draft create` now as well scaffolding a default .draftignore file. The only thing we can't move there is the .git/ repo ignore (as ** isn't supposed in helm ignore parser yet, but as `.git/` is ignored by default in git itself, I guess we can live with it for now).

I added as well some tests for `draft create`, running multiple test cases for various draft init file. You can update easily them with `-update` when changes are done in helm or the package autodetection, and add other test cases easily (I tried to separate the commits for each so that it's clear).

Hoping you will like it ;)

**Comment by bacongobbler** [comment]
One question: how are you using the `scenarios.test` files? I have never seen this convention before and would love to know more about it.

**Comment by bacongobbler** [comment]
Other than that, I will give this PR a spin.

**Comment by bacongobbler** [comment]
Seems to work as advertised.

**Comment by bacongobbler** [comment]
Oh and I completely forgot! If you have some free time, would you mind adding some documentation to [the User Guide](https://github.com/Azure/draft/blob/master/docs/user-guide.md)? If this is too much work, I would be more than happy to document it in a follow-up PR.

File: cmd/draft/create.go [hunk 2]
```diff
@@ -115,6 +115,14 @@ func (c *createCmd) run() error {
 		return fmt.Errorf("could not write metadata to draft.toml: %v", err)
 	}
 
+	ignoreFile := filepath.Join(c.dest, ignoreFileName)
+	if _, err := os.Stat(ignoreFile); os.IsNotExist(err) {
+		d1 := []byte("*.swp\n*.tmp\n*.temp\n.git*")
```
Comment (didrocks) [inline cmd/draft/create.go]: sure, added! (and adapted tests)

**Comment by didrocks** [comment]
Done! I added as well some documentation in the getting started. Let me know what you think.

On your `scenarios.test` question, I just added a simple declarative way adapated to what we wanted to check (changes, what to add/remove) so that it can be easily consummed by the shell scripts and helpers I added. I think that in the long run, the e2e tests should be switched to go to be more robust: we can continue when one is failing (right now, it was and it still stops on the first failure), do unconditional cleanup,  detect for the cluster being up first, handle timeouts in a better way. Not a priority, but I think that would be a good change, wdyt?

Keep me posted if any other changes are desired!

**Comment by bacongobbler** [comment]
Agreed. I'm not a big fan of the e2e bash script but it hobbled us on our way to a public release. I'd love to eventually rip it out with functional Go tests.

**Pull request merged**